### PR TITLE
Issue partial fix for #44: init pyproject.toml and poetry.lock

### PR DESCRIPTION
**Pull request by scottrbrtsn**
for https://github.com/open-dis/open-dis-python/issues/44

## Problem

There is a desire to become a pip package per Issue 44 linked above.  It is recommended to also include a pyproject.toml so package managers like poetry can resolve dependencies.  I recognize this is rather trivial to add for open-dis-python seeing as how (after I looked into it) there are virtually no external dependencies to create the package (only `numpy`.  Not having a `pyproject.toml` could affect applications downstream desiring to easily add and manage `opendis` as a dependency. 

However, there is concern with explicitly requiring python versions.  

## Solution
This PR initializes a pyproject.toml, a poetry lock (for the .venv) and sets the pyproject to use python 3.5 or higher.  

## Still unresolved
The issue in question could likely:
1. remove the `setup.py`
1. follow a [pypi publish guide](https://medium.com/@joel.barmettler/how-to-upload-your-python-package-to-pypi-65edc5fe9c56) or with [poetry publish](https://www.digitalocean.com/community/tutorials/how-to-publish-python-packages-to-pypi-using-poetry-on-ubuntu-22-04)
1. then update the README to simply `pip install opendis` instead of `git clone` and `pip install .`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[tool.poetry]
+name = "opendis"
+version = "0.1.0"
+description = ""
+authors = ["Scott Robertson <scott.peter.robertson@gmail.com>"]
```
Comment (scottrbrtsn) [inline pyproject.toml:5]: This was a default, we can change this to the original authors of the package.

**Comment by scottrbrtsn** [comment]
Ah, I'm using my system python... tried on fresh machine and there are dependencies:  

>The current project's supported Python range (>=3.5,<4.0) is not compatible with some of the required packages Python requirement:
  - numpy requires Python >=3.9, so it will not be satisfied for Python >=3.5,<3.9
  
 From [RangeCoordinates](https://github.com/open-dis/open-dis-python/blob/master/opendis/RangeCoordinates.py#L46)

File: pyproject.toml (hunk 1 above)
Comment (leif81) [reply pyproject.toml:5]: Yes let's change to the default authors in setup

Comment (scottrbrtsn) [reply pyproject.toml]: updated

**Comment by leif81** [comment]
> 
> ## Still unresolved
> The issue in question could likely:
> 1. remove the `setup.py`
> 1. follow a [pypi publish guide](https://medium.com/@joel.barmettler/how-to-upload-your-python-package-to-pypi-65edc5fe9c56) or with [poetry publish](https://www.digitalocean.com/community/tutorials/how-to-publish-python-packages-to-pypi-using-poetry-on-ubuntu-22-04)
> 1. then update the README to simply `pip install opendis` instead of `git clone` and `pip install .`
> 
> 

I'm good with these suggested changes as well. Let's move over values from the setup.py file into the new pyproject.toml file so they match as much as we can. Version, description, etc

**Comment by leif81** [comment]
Another good point @scottrbrtsn 
It is becoming an increasingly good case to set the baseline to Python 3.9. I support that direction now. If there any concerns please leave a comment here.

**Comment by leif81** [comment]
Looking good.

Is there any benefit to keeping the setup.py if the use of poetry is merged here?

**Comment by scottrbrtsn** [comment]
> Looking good.
> 
> Is there any benefit to keeping the setup.py if the use of poetry is merged here?

I'm not 100% on details.  Just that it would affect pip install behavior and folks might need adjust accordingly. 

Similar example:
https://github.com/wearepal/ranzen/pull/5

References:
https://github.com/python-poetry/poetry/issues/321

**Review by leif81 — APPROVED** [review]

**Pull request merged**